libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt

```diff
@@ -635,7 +631,7 @@ internal object MapboxNavigationTelemetry : MapboxNavigationTelemetryInterface {
         mapboxNavigation.unregisterOffRouteObserver(callbackDispatcher)
         mapboxNavigation.unregisterNavigationSessionObserver(navigationSessionObserver)
         Log.d(TAG, "resetting Telemetry initialization")
-        initializer = preInitializePredicate
+        weakMapboxNavigation = WeakReference(null)
```

Comment:
```suggestion
        weakMapboxNavigation.clear()
```
or 
```suggestion
        weakMapboxNavigation.set(null) 
```

Reply:
```kotlin
weakMapboxNavigation.clear()
```

TIL it seems that internally is setting the reference to `null` 🚀 

```kotlin
weakMapboxNavigation.set(null) 
```

The setter is not available for some reason 🤔

Going to go ahead and use your first suggestion 👍